### PR TITLE
IPC: port PM D3 entry fixes to CNL, WHL & CML

### PR DESCRIPTION
**Pull request by RanderWang**
The fix on APL is at: https://github.com/thesofproject/sof/pull/772/files
On CNL, we need to set two registers: IPC_DIPCTDR and IPC_DIPCTDA

This fixes https://github.com/thesofproject/sof/issues/773

### CODE REVIEW

File: src/platform/cannonlake/power_down.S [hunk 1]
```diff
@@ -144,6 +144,21 @@ beqz pu32_hpsram_mask, _PD_SLEEP
 	m_cavs_set_hpldo_state temp_reg0, temp_reg1, temp_reg2
 
 //TODO: add sending IPC reply from L1$ locked code
+_PD_SEND_IPC:
+/* Send IPC to host informing of PD completion - Clear BUSY
+ * bit by writing IPC_DIPCTDR_BUSY to IPC_DIPCTDR
+ * and writing IPC_DIPCTDA_BUSY to IPC_DIPCTDA
+ */
+	movi temp_reg0, IPC_HOST_BASE
+	l32i temp_reg1, temp_reg0, IPC_DIPCTDR
```
Comment (lgirdwood) [inline src/platform/cannonlake/power_down.S]: @RanderWang just to confirm this was based on APL version and modified for differences in CNL IPC registers ?

**Comment by lgirdwood** [comment]
@lech @mmaka1 any comments ?

File: src/platform/cannonlake/power_down.S (hunk 1 above)
Comment (RanderWang) [reply src/platform/cannonlake/power_down.S]: yes, It is based on APL version and I change the registers for CNL

**Comment by lgirdwood** [comment]
@RanderWang please check internal CI if no update in a few hours and let me know if it passes.

**Pull request merged**